src/shaders/fill_extrusion_pattern.fragment.glsl

```diff
@@ -38,7 +43,18 @@ void main() {
     vec4 out_color = mix(color1, color2, u_fade);
 
     out_color = out_color * v_lighting;
-
+#ifdef FAUX_AO
+    float intensity = 1.0 - sqrt(1.0 - u_ao[0]);
+    float h = max(0.0, v_ao.z);
+    float h_floors = h / u_ao[1];
+    float y_shade = mix(1.0, 0.1, intensity * min(v_ao.y, 1.0));
+    float shade = mix(1.0, 0.92, intensity) * (y_shade + (1.0 - y_shade) * (1.0 - pow(1.0 - min(h_floors / 16.0, 1.0), 16.0))) + 0.08 * intensity * min(h_floors / 160.0, 1.0);
+    // concave angle
+    float concave = v_ao.x * v_ao.x;
+    float x_shade = mix(1.0, mix(0.6, 0.75, min(h_floors / 30.0, 1.0)), intensity) + intensity * mix(0.0, 0.1, min(h, 1.0));
```

Comment:
It's a bit extraneous to write `mix` statements with constants, why not inline to a simpler form, such as `0.1 * min(h, 1.0)` instead of `mix(0.0, 0.1, min(h, 1.0))`. This could apply to the other `mix` statements above as well.